src/link/Elf.zig

```diff
@@ -2480,7 +2480,7 @@ fn linkWithLLD(self: *Elf, arena: Allocator, prog_node: *std.Progress.Node) !voi
             try argv.append("-pie");
         }
 
-        if (is_dyn_lib and target.os.tag == .netbsd) {
+        if (is_exe_or_dyn_lib and target.os.tag == .netbsd) {
```

Comment:
I haven't yet run into a situation where an executable is created with > 2 PT_LOAD. Is this easily reproduced?

Reply:
Yes, zig cc -o main main.c produce 4 PT_LOAD.

```
$ zig/build/stage3/bin/zig version          
0.12.0-dev.3653+e45bdc6bd
$ zig/build/stage3/bin/zig cc -o main main.c
$ readelf -l main

Elf file type is EXEC (Executable file)
Entry point 0x1001840
There are 11 program headers, starting at offset 64

Program Headers:
  Type           Offset             VirtAddr           PhysAddr
                 FileSiz            MemSiz              Flags  Align
  PHDR           0x0000000000000040 0x0000000001000040 0x0000000001000040
                 0x0000000000000268 0x0000000000000268  R      0x8
  INTERP         0x00000000000002a8 0x00000000010002a8 0x00000000010002a8
                 0x0000000000000017 0x0000000000000017  R      0x1
      [Requesting program interpreter: /usr/libexec/ld.elf_so]
  LOAD           0x0000000000000000 0x0000000001000000 0x0000000001000000
                 0x0000000000000840 0x0000000000000840  R      0x1000
  LOAD           0x0000000000000840 0x0000000001001840 0x0000000001001840
                 0x00000000000004a0 0x00000000000004a0  R E    0x1000
  LOAD           0x0000000000000ce0 0x0000000001002ce0 0x0000000001002ce0
                 0x00000000000001e0 0x00000000000001e0  RW     0x1000
  LOAD           0x0000000000000ec0 0x0000000001003ec0 0x0000000001003ec0
                 0x0000000000000052 0x0000000000000069  RW     0x1000
  DYNAMIC        0x0000000000000d08 0x0000000001002d08 0x0000000001002d08
                 0x0000000000000140 0x0000000000000140  RW     0x8
  GNU_RELRO      0x0000000000000ce0 0x0000000001002ce0 0x0000000001002ce0
                 0x00000000000001e0 0x0000000000000320  R      0x1
  GNU_EH_FRAME   0x00000000000006c8 0x00000000010006c8 0x00000000010006c8
                 0x0000000000000044 0x0000000000000044  R      0x4
  GNU_STACK      0x0000000000000000 0x0000000000000000 0x0000000000000000
                 0x0000000000000000 0x0000000001000000  RW     0x0
  NOTE           0x00000000000002c0 0x00000000010002c0 0x00000000010002c0
                 0x000000000000002c 0x000000000000002c  R      0x4

 Section to Segment mapping:
  Segment Sections...
   00     
   01     .interp 
   02     .interp .note.netbsd.ident .note.netbsd.pax .dynsym .gnu.hash .hash .dynstr .rela.dyn .rela.plt .rodata .eh_frame_hdr .eh_frame 
   03     .text .init .fini .plt 
   04     .ctors .dtors .jcr .dynamic .got .got.plt 
   05     .data .bss 
   06     .dynamic 
   07     .ctors .dtors .jcr .dynamic .got .got.plt 
   08     .eh_frame_hdr 
   09     
   10     .note.netbsd.ident .note.netbsd.pax 
```

Reply:
can you provide `main.c`? I am unable to reproduce with a simple hello world.

Reply:
ok I can reproduce. The sad part is that a > 2 PT_LOAD exe still executes. But it seems `ldd` is the only thing reporting the error.  On the one hand it's clear that .so need the flags, but it's not 100% that an exe needs the flags, because they still execute.

May I ask, under what use case did you happen that > 2 PT_LOAD exe is bugged?

Reply:
I just use ldd usually instead of readelf and gcc/clang produce only 2 PT_LOAD on NetBSD. Nothing else really important.

Reply:
I think I broke my netbsd branch. Should I make another PR for the second part of this change ?

Reply:
> I just use ldd usually instead of readelf and gcc/clang produce only 2 PT_LOAD on NetBSD. Nothing else really important.

fyi, the > 2 PT_LOAD is caused by llvm's linker. This is why both gcc/clang produce 2 PT_LOAD, they're both using system linker.

I'm ok with the 2nd part of your patch.

Reply:
> I think I broke my netbsd branch. Should I make another PR for the second part of this change ?

You can tweak your local branch and get it into shape. If things go really wrong, I'd suggest create a "save branch".... so in your local branch, `git branch foobar` then `git reset --hard` the branch to anything you want, cherry-pick the commit(s) from foobar and once you're happy, force-push with `git push -f <remote-name-of-github>` . That should fix things up.

Reply:
Ok, I squashed the change to 1 commit as suggested bv @mikdusan, thank you for your help with git.